### PR TITLE
Add BASE_DOCKER_REGISTRY_CREDENTIAL_ID for external-test

### PR DESCRIPTION
**Pull request by LongyuZhang**
- BASE_DOCKER_REGISTRY_CREDENTIAL_ID is needed to increase daily image pull limit when running container-based external testing
- Related Issue: runtimes/backlog/issues/1234

### CODE REVIEW

**Comment by LongyuZhang** [comment]
@AdamBrousseau Could you review it? Thanks.

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -414,10 +414,14 @@ class Build {
 
                         def DOCKER_REGISTRY_URL = ''
                         def DOCKER_REGISTRY_URL_CREDENTIAL_ID = ''
+                        def BASE_DOCKER_REGISTRY_CREDENTIAL_ID = ''
                         if ("${testType}".contains('external')) {
                             DOCKER_REGISTRY_URL = 'docker-na.artifactory.swg-devops.com/sys-rt-docker-local'
                             DOCKER_REGISTRY_URL_CREDENTIAL_ID = artifactoryCredential
                             rerunIterations = '0'
+                            if (buildConfig.DOCKER_CREDENTIAL) {
```
Comment (AdamBrousseau) [inline pipelines/build/common/openjdk_build_pipeline.groovy:422]: DOCKER_CREDENTIAL currently is used for Docker based compiles in order to authenticate to docker.io to pull containers. At the moment that is only a/x linux. We have a TODO to expand that to p/z. Does the external tests already run on z/p? If so, we'll likely need to add the cred to those specs even before the compiles are converted over. If we do add the cred, we'll have to check the code doesn't have anymore `if` conditions that may get executed if we add the cred without the rest of the docker compile spec info.

Comment (AdamBrousseau) [reply pipelines/build/common/openjdk_build_pipeline.groovy:422]: I'm answering my own question. External tests run on z/p already
https://github.com/ibmruntimes/ci-jenkins-pipelines/blob/1ee9b08642869815af4bb7fc8d65ed713adbba69/pipelines/jobs/configurations/jdk11u_pipeline_config.groovy#L401
https://github.com/ibmruntimes/ci-jenkins-pipelines/blob/1ee9b08642869815af4bb7fc8d65ed713adbba69/pipelines/jobs/configurations/jdk11u_pipeline_config.groovy#L433
And we have code in `openjdk_build_pipeline.groovy`
https://github.com/ibmruntimes/ci-jenkins-pipelines/blob/1ee9b08642869815af4bb7fc8d65ed713adbba69/pipelines/build/common/openjdk_build_pipeline.groovy#L2223
https://github.com/ibmruntimes/ci-jenkins-pipelines/blob/1ee9b08642869815af4bb7fc8d65ed713adbba69/pipelines/build/common/openjdk_build_pipeline.groovy#L2268
However, I see those are nested within an `if (DOCKER_IMAGE)` so I think we are ok to just add the cred to those specs.
https://github.com/ibmruntimes/ci-jenkins-pipelines/blob/1ee9b08642869815af4bb7fc8d65ed713adbba69/pipelines/build/common/openjdk_build_pipeline.groovy#L2188

**Review by AdamBrousseau — CHANGES_REQUESTED** [review]
So I think if you add the `dockerCredential` lines to the specs you are running External tests then this PR should be fine.

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (LongyuZhang) [reply pipelines/build/common/openjdk_build_pipeline.groovy:422]: Yes, we run docker-based testing on all a/x/z/p linux. But in both this repo and the test repo, we use `if` condition for this parameter, so though we only have it on a/x currently, we can reduce some anonymous image pulling on these platforms to have more quotas on other platforms then.

**Comment by LongyuZhang** [comment]
`dockerCredential` lines have been added to z/p linux on JDK 11, 17 and 21 as suggested. Thanks.

> So I think if you add the `dockerCredential` lines to the specs you are running External tests then this PR should be fine.

**Review by AdamBrousseau — APPROVED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -523,6 +527,7 @@ class Build {
                                             context.booleanParam(name: 'DYNAMIC_COMPILE', value: DYNAMIC_COMPILE),
                                             context.string(name: 'DOCKER_REGISTRY_URL', value: DOCKER_REGISTRY_URL),
                                             context.string(name: 'DOCKER_REGISTRY_URL_CREDENTIAL_ID', value: DOCKER_REGISTRY_URL_CREDENTIAL_ID),
+                                            context.string(name: 'BASE_DOCKER_REGISTRY_URL_CREDENTIAL_ID', value: BASE_DOCKER_REGISTRY_URL_CREDENTIAL_ID),
```
Comment (llxia) [inline pipelines/build/common/openjdk_build_pipeline.groovy:530]: `BASE_DOCKER_REGISTRY_CREDENTIAL_ID` is set above, but we are using `BASE_DOCKER_REGISTRY_URL_CREDENTIAL_ID` here (which is not being defined)

**Pull request merged**